### ISSUE TITLE
major bugs report with model.evaluate() and model.predict()

### ISSUE THREAD

**Issue by YUX** [apple/tensorflow_macos#145]
I tried a basic Keras fashion mnist example on both m1 and colab, and I found that **model.evaluate()** and **model.predict()** on m1 are just wrong.

**The accuracy on test set aound  0.8477 is normal (Colab), but 0.1006 doesn't make any sense (m1).**

Code:

 ```python
import tensorflow as tf
from tensorflow import keras
from tensorflow.keras.layers import Flatten, Dense
from tensorflow.keras.models import Sequential

fashion_mnist = keras.datasets.fashion_mnist
(X_train_full,y_train_full),(X_test,y_test) = fashion_mnist.load_data()
X_valid,X_train = X_train_full[:5000] / 255.,X_train_full[5000:]/255.
y_valid,y_train = y_train_full[:5000], y_train_full[5000:]

model = Sequential([
    Flatten(input_shape=[28,28]),
    Dense(300,activation="relu"),
    Dense(100,activation="relu"),
    Dense(10,activation="softmax")
])

model.compile(loss="sparse_categorical_crossentropy",
             optimizer="sgd",
             metrics=["accuracy"])

history = model.fit(X_train,y_train,epochs=30,
                   validation_data=(X_valid,y_valid))

model.evaluate(X_test,y_test)

X_new = X_test[:3]
y_proba = model.predict(X_new)
y_proba
```
---
### the m1 output :
<img width="1110" alt="CleanShot 2021-01-29 at 18 37 30@2x" src="https://user-images.githubusercontent.com/10670661/106264784-15154f80-6261-11eb-9e4f-f3b41ec18e09.png">

---
### the colab output:
<img width="1197" alt="CleanShot 2021-01-29 at 18 37 59@2x" src="https://user-images.githubusercontent.com/10670661/106264837-22cad500-6261-11eb-891e-be5e82947b73.png">

**Comment by anna-tikhonova** [comment]
Thank you very much for reporting this and providing a reproducible test case. We will investigate and report back.

**Comment by icenando** [comment]
@anna-tikhonova 
Hi Anna,

I'm experiencing the same issue. Please see `model.predict` results below. On the left it's my results, on the right it's Kaggle's:

<img width="503" alt="Screenshot 2021-02-07 at 17 39 11" src="https://user-images.githubusercontent.com/12436042/107154688-4b8f5080-696c-11eb-9d14-d1ef25db3e7f.png">

Here is a link to my Kaggle notebook, so that you have the full code: [https://www.kaggle.com/nandomachado/test-notebook](url)

Also, I seeded my `random`, `np.random` and `tf.random.set_seed` . All results, including weights are identical between my computer and Kaggle's up to when I call `model.predict`. 

The code is also the same: I simply uploaded it to Kaggle with no changes.

**Comment by ashkan-76** [comment]
I have an identical problem. I run a conv-net and the loss & val_loss look solid, but predict returns garbage.
I think you're already aware of the LSTM issue.

**Comment by xxlbaslxx** [comment]
I'm here for a same problem. I ran a same code between my M1 and google colab but the result are totally different.
<img width="1440" alt="Screen Shot 2564-02-14 at 15 44 08" src="https://user-images.githubusercontent.com/47595178/109041398-8d0c5380-7701-11eb-9cf9-1c227e330f3a.png">
The confusion matrix on my jupyter look too weird to possible. 
PS. I also use **model.predict** on my code too.

**Comment by pelayocampa** [comment]
Same here. Doing tests with this https://colab.research.google.com/github/tensorflow/docs-l10n/blob/master/site/es-419/tutorials/keras/classification.ipynb#scrollTo=E51yS7iCCaXO example.
M1 doesn't predict a single piece of clothing :)

**Comment by maralski** [comment]
The problem appears to be the use of validation_data in fit(). Using this impacts predict(). Perform validation outside of fit() and the results should be good.

**Comment by Arfius** [comment]
are you using Rosetta2?

**Comment by ashkan-76** [comment]
No, all native M1.

**Comment by Arfius** [comment]
I have same problem using Rosetta2 as well....

**Comment by tiatordos** [comment]
![image](https://user-images.githubusercontent.com/10973788/117330813-5d629000-ae96-11eb-82fb-9eb53b2738a3.png)

worst , changing the number os items selected changes the prediction
